src/languages/python.js

```diff
@@ -194,12 +194,36 @@ export default function(hljs) {
     ]
   };
 
+  // https://docs.python.org/3/reference/lexical_analysis.html#numeric-literals
+  const digitpart = '[0-9](_?[0-9])*';
+  const pointfloat = '(dp)?\\.(dp)|(dp)\\.'.replace(/dp/g, digitpart);
   const NUMBER = {
     className: 'number', relevance: 0,
     variants: [
-      {begin: hljs.BINARY_NUMBER_RE + '[lLjJ]?'},
-      {begin: '\\b(0o[0-7]+)[lLjJ]?'},
-      {begin: hljs.C_NUMBER_RE + '[lLjJ]?'}
+      // exponentfloat, pointfloat
+      // https://docs.python.org/3/reference/lexical_analysis.html#floating-point-literals
+      // optionally imaginary
+      // https://docs.python.org/3/reference/lexical_analysis.html#imaginary-literals
+      {begin: '((dp)|(pf))[eE][+-]?(dp)[jJ]?'
+        .replace(/dp/g, digitpart).replace(/pf/g, pointfloat)},
+      {begin: '(pf)[jJ]?'
+        .replace(/pf/g, pointfloat)},
+
+      // decinteger, bininteger, octinteger, hexinteger
+      // https://docs.python.org/3/reference/lexical_analysis.html#integer-literals
+      // optionally "long" in Python 2
+      // https://docs.python.org/2.7/reference/lexical_analysis.html#integer-and-long-integer-literals
+      // decinteger is optionally imaginary
+      // https://docs.python.org/3/reference/lexical_analysis.html#imaginary-literals
+      {begin: '\\b([1-9](_?[0-9])*|0+(_?0)*)[lLjJ]?\\b'},
+      {begin: '\\b0[bB](_?[01])+[lL]?\\b'},
+      {begin: '\\b0[oO](_?[0-7])+[lL]?\\b'},
+      {begin: '\\b0[xX](_?[0-9a-fA-F])+[lL]?\\b'},
+
+      // imagnumber (digitpart-based)
+      // https://docs.python.org/3/reference/lexical_analysis.html#imaginary-literals
+      {begin: '\\b(dp)[jJ]\\b'
+        .replace(/dp/g, digitpart)},
```

Comment:
```suggestion
      {begin: `\\b(${digitpart})[jJ]\\b`
```

Lets use template literals for all these and loose the replace stuff.

Reply:
Cool, I wasn't sure if they were welcome in the primary code. Done.